tensorflow/c/tf_status.h

```diff
@@ -87,6 +87,12 @@ TF_CAPI_EXPORT extern TF_Code TF_GetCode(const TF_Status* s);
 // TF_OK.
 TF_CAPI_EXPORT extern const char* TF_Message(const TF_Status* s);
 
+// For propagating errors when calling a function.
```

Comment:
Let's move this to the start of the file, like the other macros.

Also, it should return the status.

Reply:
I add a return value so we can choose what should be returned. Actually, in some function it will return -1. We could ask a `printf` to print the message of the `status`. Slient failure won't happen even without `printf` if we use this macro with filesystem_plugins since `modular_filesystem.cc` will be the ones who check that `status`

Reply:
Let me first review where it gets used to ensure that the added complexity is ok for a new macro.